packages/compass-shell/src/stores/store.js

```diff
@@ -19,27 +19,32 @@ export default class CompassShellStore {
 
     this.globalAppRegistry = appRegistry;
 
-    appRegistry.on(
-      'data-service-connected',
-      this.onDataServiceConnected
-    );
-
-    appRegistry.on(
-      'data-service-disconnected',
-      this.onDataServiceDisconnected
+    appRegistry.on('data-service-connected', this.onDataServiceConnected);
+
+    appRegistry.on('data-service-disconnected', this.onDataServiceDisconnected);
+
+    setupLoggerAndTelemetry(
+      appRegistry,
+      createLogger('COMPASS-SHELL').log.unbound,
+      () => {
+        throw new Error('no analytics integration for compasss-shell');
+      },
+      {
+        platform: process.platform,
+        arch: process.arch,
+      },
+      require('../../package.json').version
```

Comment:
I don't think we need to do anything now, but on the mongosh logger side of this the variable received here is called `mongosh_version` which might be a bit misleading in telemetry since it's really the `compass-shell` version that we'll be showing here. Would it make sense to have both the consumer (compass-shell) version and the mongosh version here?
https://github.com/mongodb-js/mongosh/blob/main/packages/logging/src/setup-logger-and-telemetry.ts#L82

Reply:
So … this argument is only used for telemetry (which we don’t use here), not for logging, so I was thinking that it’s fine to just use the next best thing.

The problem is that the only consistent “mongosh” version that we definitely have is the CLI package’s version, which may also just end up not applying anymore here if we ever decide to use lerna independent mode in mongosh. I’ve added logging of the service-provider-server package version on the mongosh side, right now that’s in sync with the CLI package, but I’ve also warned Felicia that they might diverge at some point. :)